app/models/user.rb

```diff
@@ -1,4 +1,15 @@
 class User < ApplicationRecord
   validates :name, presence: true
   validates :email, presence: true
+
+  def self.create_from_omniauth(auth)
+    # Creates a new user only if it doesn't exist
+    where(email: auth.info.email).first_or_initialize do |user|
```

Comment:
There's also [`find_or_create_by`](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-find_or_create_by), perhaps it makes things a bit shorter.